md-dao/src/main/java/com/wipro/ats/bdre/md/dao/ProcessTemplateDAO.java

```diff
@@ -82,20 +86,17 @@
                     processTemplateList.add(processTemplate);
                 }
 
-            } else if (fetchProcessTPPId.list().size() == 0 && fetchProcessTPId.list().size() == 1) {
-// (select count(*) from process_template where parent_process_id =pid)=0 and (select count(*) from process_template where process_template_id=pid) =1
-
-//where (process_template_id = ppid and delete_flag != 1 and process_template_id !=0) order by process_template_id limit limit_id offset page  ;
+            } else if (fetchProcessTPPId.list().isEmpty() && fetchProcessTPId.list().size() == 1) {
                 ProcessTemplate processTemplatePid = (ProcessTemplate) fetchProcessTPId.uniqueResult();
                 if (processTemplatePid.getProcessTemplate() != null && processTemplatePid != null) {
                     Integer parentProcessId = processTemplatePid.getProcessTemplate().getProcessTemplateId();
 
-                    Criteria fetchProcessTemplate = session.createCriteria(ProcessTemplate.class).add(Restrictions.eq("processTemplateId", parentProcessId)).add(Restrictions.eq("deleteFlag", false)).add(Restrictions.not(Restrictions.eq("processTemplateId", 0))).addOrder(Order.asc("processTemplateId"));
+                    Criteria fetchProcessTemplate = session.createCriteria(ProcessTemplate.class).add(Restrictions.eq(PROCESSTEMPLATEID, parentProcessId)).add(Restrictions.eq(DELETEFLAG, false)).add(Restrictions.not(Restrictions.eq(PROCESSTEMPLATEID, 0))).addOrder(Order.asc(PROCESSTEMPLATEID));
                     fetchProcessTemplate.setFirstResult(pageNum);
                     fetchProcessTemplate.setMaxResults(numResults);
                     List<ProcessTemplate> jpaPTList = fetchProcessTemplate.list();
 
-                    LOGGER.info("List size of process template :" + fetchProcessTemplate.list().size());
+                    LOGGER.info("List size of process template : " + fetchProcessTemplate.list().size());
                     for (ProcessTemplate jpaProcessTemplate : jpaPTList) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)